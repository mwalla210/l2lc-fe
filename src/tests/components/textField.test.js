import React from 'react'
import renderer from 'react-test-renderer'
import TextField from '../../components/textField'

describe('TextField', () => {
  it ('Renders with valid false', () => {
    let options = {
      page: {},
      id: '',
      disabled: false,
      value: '',
      onChange: jest.fn(),
      onBlur: jest.fn(),
      valid: false,
      focus: false
    }
    const component = renderer.create(
      <TextField {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with valid true', () => {
    let options = {
      page: {},
      id: '',
      disabled: false,
      value: '',
      onChange: jest.fn(),
      onBlur: jest.fn(),
      valid: true,
      focus: false
    }
    const component = renderer.create(
      <TextField {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
