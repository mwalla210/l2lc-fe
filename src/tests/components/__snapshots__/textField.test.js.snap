// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Renders with valid false 1`] = `
<input
  autoFocus={false}
  className="form-control"
  disabled={false}
  id=""
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  style={
    Object {
      "borderColor": "orange",
    }
  }
  type="text"
  value=""
/>
`;

exports[`TextField Renders with valid true 1`] = `
<input
  autoFocus={false}
  className="form-control"
  disabled={false}
  id=""
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  style={
    Object {
      "borderColor": "",
    }
  }
  type="text"
  value=""
/>
`;
